airflow/www/api/experimental/endpoints.py

```diff
@@ -44,16 +49,35 @@ def trigger_dag(dag_id):
     if 'conf' in data:
         conf = data['conf']
 
+    execution_date = None
+    if 'exec_date' in data:
+        execution_date = data['exec_date']
+
+        # Convert string datetime into actual datetime
+        try:
+            execution_date = datetime.strptime(execution_date,
+                                               '%Y-%m-%dT%H:%M:%S')
```

Comment:
I don't think this is ISO 8601 compliant: https://en.wikipedia.org/wiki/ISO_8601